### PR TITLE
Added mute-audio setup options

### PR DESCRIPTION
**Pull request by moshe-braner**
allow muting audio while in setup, in sink, or generally (with or without muting alarms)

### CODE REVIEW

File: main/SetupMenu.cpp [hunk 1]
```diff
@@ -980,12 +980,44 @@ void SetupMenu::audio_menu_create_volume( MenuEntry *top ){
...
 }
 
+void SetupMenu::audio_menu_create_mute( MenuEntry *top ){
+	SetupMenuSelect * asida = new SetupMenuSelect( PROGMEM"In Sink", RST_NONE, 0 , true, &audio_mute_sink );
+	asida->setHelp(PROGMEM"Select whether vario audio will be muted while in sink");
+	asida->addEntry( PROGMEM"Stay On");  // 0
+	asida->addEntry( PROGMEM"Mute");     // 1
+	top->addEntry( asida );
+
+	SetupMenuSelect * ameda = new SetupMenuSelect( PROGMEM"In Setup", RST_NONE, 0 , true, &audio_mute_menu );
```
Comment (iltis42) [inline main/SetupMenu.cpp:990]: guess this is redundant to already defined "audio_disable" variable, why we need a second setup element here?
Note by renaming this, an in service upgrade will fail, pilots using already the above will fall back to the new default, even they had configured it.

File: main/SetupMenu.cpp [hunk 2]
```diff
@@ -980,12 +980,44 @@ void SetupMenu::audio_menu_create_volume( MenuEntry *top ){
...
+	top->addEntry( ameda );
+
+	SetupMenuSelect * ageda = new SetupMenuSelect( PROGMEM"Generally", RST_NONE, 0 , true, &audio_mute_gen );
+	ageda->setHelp(PROGMEM"Select audio on, or vario audio muted, or all audio muted including alarms");
+	ageda->addEntry( PROGMEM"Audio On");      // 0
+	ageda->addEntry( PROGMEM"Alarms On");     // 1
+	ageda->addEntry( PROGMEM"Audio Off");     // 2
+	top->addEntry( ageda );
+
+	SetupMenuSelect * amps = new SetupMenuSelect( PROGMEM"Amplifier", RST_NONE, 0 , true, &amplifier_shutdown );
```
Comment (iltis42) [inline main/SetupMenu.cpp:1003]: How about the old mode of shutdown immediate? 
Note there is old hardware in field with some sizzling noise when this doesn't take place immediate. 
Propose a third option:
0 Stay On
1 Shutdown        -> immediate shutdown, old behavior
2 Idle Shutdown -> new: shutdown during long silence

File: main/SetupNG.h [hunk 3]
```diff
@@ -615,7 +615,9 @@ extern SetupNG<int> 		wk_label_minus_1;
 extern SetupNG<int> 		wk_label_minus_2;
 extern SetupNG<int> 		wk_label_minus_3;
 extern SetupNG<float>       flap_takeoff;
-extern SetupNG<int> 		audio_disable;
+extern SetupNG<int> 		audio_mute_menu;  // replaced audio_disable (same key)
```
Comment (iltis42) [inline main/SetupNG.h:618]: Ahh i see, same key, so this should work with older sw loads for in service upgrade.

File: main/ESPAudio.cpp [hunk 4]
```diff
@@ -600,9 +600,22 @@ void Audio::dactask(void* arg )
...
+					// Optionally disable vario audio when in Sink
+					if( audio_mute_sink.get() && _te < 0 )
+						sound = false;
+					// Optionally disable vario audio when in setup menu
+					else if( audio_mute_menu.get() && gflags.inSetup )
+						sound = false;
+					// Optionally disable vario audio generally
+					else if( audio_mute_gen.get() != 0 )  // == 1 or 2
+						sound = false;
+				} else if( audio_mute_gen.get() == 2 ) {
```
Comment (iltis42) [inline main/ESPAudio.cpp:614]: Instead of '0' '1'  '2' better use an enum here for clarity.

**Comment by moshe-braner** [comment]
Added enums, amplifier shutdown is immediate.  The option to shutdown only after 5 seconds of silence was not actually included in this PR, thus mentioning it in the Setup Menu help text was an error.  The option of delayed amplifier shutdown after 5 seconds of silence will be added in the next (RICO style audio) PR.

File: main/ESPAudio.cpp [hunk 5]
```diff
@@ -680,8 +696,13 @@ void Audio::dactask(void* arg )
 					}
 					dacDisable();
 				}
-				if( deadband_active && amplifier_shutdown.get() )
+				if( shutdownamp ) {
+					if( deadband_active 
+					 || ( audio_mute_sink.get() && _te < 0 )
+					 || ( audio_mute_menu.get() && gflags.inSetup )
+					 || ( audio_mute_gen.get() != AUDIO_ON ) )
 					enableAmplifier( false );
```
Comment (moshe-braner) [inline main/ESPAudio.cpp:704]: This repeating of the silencing conditions is not a good implementation.  (And not quite correct regarding general muting during alarms.)  My intention is to submit the RICO audio style version, which also has a better implementation of this detail, using a variable that flags the intention to shut down the amplifier.

**Comment by moshe-braner** [comment]
Closing this PR, will submit a revised one with the flag variable for shutting down amp.

**Pull request closed without merging**